extensions/amp-user-notification/0.1/amp-user-notification.js

```diff
@@ -0,0 +1,279 @@
+      method: 'POST'
+    });
+  }
+
+  /**
+   * Success handler for `fetchShowEndpoint_`
+   * @param {GetResponseMetadata}
+   * @private
+   */
+  onFetchShowEndpointSuccess_(data) {
+    assert(typeof data.show_notification == 'boolean', '%s should ' +
+        'be a boolean', data.show_notification);
+    if (data.show_notification) {
+      this.show();
+    }
+    return data;
+  }
+
+  /**
+   * Generic rejection handler for amp-user-notification.
+   * @private
+   */
+  onRejection_(msg) {
+    console/*OK*/.log('Rejection ocurred on `amp-user-notification` ' +
+        `with html element id "${this.id_}".`);
+  }
+
+  /**
+   * Get cid from service.
+   * @private
+   */
+  getAsyncCid_() {
+    return cidFor(this.win_).then((cid) => {
+      return cid.get('amp-user-notification',
+          Promise.resolve(), this.dialogPromise_);
+    });
+  }
+
+  /**
+   * Shows the current user notification. Adds the `.amp-active` class to
+   * the element.
+   * @export
+   */
+  show() {
+    this.element.classList.add('amp-active');
+  }
+
+  /**
+   * Hides the current user notification and invokes the `dialogResolve_`
+   * method. Removes the `.amp-active` class from the element.
+   * @export
+   */
+  dismiss() {
+    this.vsync_.mutate(() => {
+      this.element.classList.remove('amp-active');
+      this.element.classList.add('amp-hidden');
+      this.dialogResolve_();
+    });
+  }
+}
+
+
+class UserNotification {
+
+  constructor(window) {
+    this.win_ = window;
+
+    /** @private @const {!Object<UserNotificationDefer>} */
+    this.deferRegistry_ = Object.create(null);
+
+    this.glasspane = this.win_.document.createElement('div');
+    this.glasspane.classList.add('-amp-glasspane');
+    this.win_.document.body.appendChild(this.glasspane);
+  }
+
+  /**
+   * @return {!Promise|null}
+   */
+  get(id) {
+    let defer = this.getElementDeferById_(id);
+    if (defer) {
+      return defer.promise;
+    }
+    return null;
+  }
+
+  /**
+   * @param {string|undefined} id
+   * @return boolean
+   */
+  has(id) {
+    if (!id) {
+      return false;
+    }
+    return !!this.getElementDeferById_(id);
+  }
+
+  registerUserNotification(id, consentPromise) {
+    let deferred = this.getElementDeferById_();
+    let resolve_;
+
+    assert(this.getElementDeferById_(id) == undefined, 'Duplicate ' +
+        'amp-use-notification id found.');
+
+    const promise = new Promise((resolve, reject) => {
+      resolve_ = resolve;
+    });
+
+    deferred = this.deferRegistry_[id] = {
+      promise: promise,
+      resolve: resolve_
+    }
+    deferred.resolve(consentPromise);
+  }
+
+  getElementDeferById_(id) {
+    return this.deferRegistry_[id];
+  }
+}
+
+/**
+ * @private
+ * @return {!UserNotification}
+ */
+function getUserNotification_(window) {
+  return getService(window, 'userNotification', () => {
+    return new UserNotification(window);
+  });
+}
+
+export function installUserNotification(window) {
```

Comment:
done. removed the indirect `getUserNotification_` method and moved the `getService` call in the install method.